inginious/frontend/pages/course_admin/utils.py

```diff
@@ -296,19 +296,22 @@ def make_csv(data):
             rval[k] = v
 
     # Convert everything to CSV
-    columns = set()
     output = [[]]
-    if isinstance(data, dict):
-        output[0].append("id")
-        for entry in data:
-            for col in data[entry]:
-                columns.add(col)
+    if lighten:
+        columns = ["_id","best","courseid","grade","result","status","submitted_on","taskid","username"]
```

Comment:
This goes against the principles of the function code, which remains content agnostic. This is one of the things we should be careful doing Python code. Things hardcoded here will clearly be missed in case of refactor (changing field names, ...).

A more robust solution should be dynamic. Something even more convenient would be to select the displayable columns in the frontend UI and send the columns ids from the form. 

Another valid and more straightforward solution, as what is called here "lightened csv" is to provide the current visible columns only, is simply to make the form send back the shown columns ids.